var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -212,6 +213,9 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("kokkos-nvcc-wrapper@master", when="@master+wrapper")
     conflicts("+wrapper", when="~cuda")
 
+    conflicts("+rocthrust", when="@4.3: ~rocm")
+    conflicts("~rocthrust", when="@4.3: +rocm")
```

Comment:
I am not so familiar with Spack but isn't registering the variant and doing this just a convoluted way of doing `depends_on("rocthrust", when="@4.3+rocm")` in the end?

Reply:
i put that to address both the below- 
spack install kokkos + rocm +rocthrust
spack install kokkos +rocm